test/test_runner/test_runner.cpp

```diff
@@ -172,6 +180,45 @@ void TestRunner::checkPlanResult(Connection& conn, QueryResult* result, TestStat
     }
 }
 
+void TestRunner::writeOutput(QueryResult* result, TestStatement* statement, size_t resultIdx) {
+    TestQueryResult& testAnswer = statement->result[resultIdx];
+    statement->testResultType = testAnswer.type;
+    std::string newOutput;
+    switch (testAnswer.type) {
+    case ResultType::OK: {
+        newOutput +=
+            "---- " + (result->isSuccess() ? std::string("ok") : std::string("error")) + '\n';
+        if (!result->isSuccess())
+            newOutput += result->getErrorMessage() + '\n';
+    } break;
+    case ResultType::HASH: {
+        std::string resultHash = convertResultToMD5Hash(*result, statement->checkOutputOrder,
+            statement->checkColumnNames);
+        newOutput += resultHash;
+    } break;
+    case ResultType::TUPLES: {
+        newOutput += "---- " + std::to_string(result->getNumTuples()) + '\n';
+        std::vector<std::string> resultTuples = convertResultToString(*result,
+            statement->checkOutputOrder, statement->checkColumnNames);
+        for (auto result : resultTuples) {
```

Comment:
nit: `result` -> `res` or something to avoid conflict with the shadowed `result`. Okay here but avoids potential errors in the future.